assembl/static2/js/app/components/debate/thread/nuggets.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import range from 'lodash/range';
+
+import { getDomElementOffset, setDomElementOffset } from '../../../utils/globalFunctions';
+
+class Nuggets extends React.Component {
+  static get SPACER_SIZE() {
+    // Vertical space between overflowing extracts in px
+    return 100;
+  }
+
+  static completeLevelFromId(id) {
+    return id.split('-').slice(1).map((ident) => {
+      return Number(ident);
+    });
+  }
+
+  static completeLevel(rowIndex, fullLevel) {
+    return fullLevel ? [rowIndex, ...fullLevel] : [rowIndex];
+  }
+
+  static completeLevelGreater(a, b) {
+    return range(Math.min(a.length, b.length)).reduce((result, index) => {
+      if (result !== null) return result;
+      if (a[index] === b[index] && (a.length === index + 1 || b.length === index + 1)) return a.length > b.length;
+      return a[index] !== b[index] ? a[index] > b[index] : result;
+    }, null);
+  }
+
+  static previousExtract(completeLevel, otherExtracts) {
+    return !otherExtracts || !otherExtracts.length
+      ? null
+      : otherExtracts.reduce((prevExtract, extract) => {
+        const currentCompleteLevel = prevExtract ? Nuggets.completeLevelFromId(prevExtract.getAttribute('id')) : null;
+        const otherCompleteLevel = Nuggets.completeLevelFromId(extract.getAttribute('id'));
+        if (
+          Nuggets.completeLevelGreater(completeLevel, otherCompleteLevel) &&
+            (prevExtract === null || Nuggets.completeLevelGreater(otherCompleteLevel, currentCompleteLevel))
+        ) {
+          return extract;
+        }
+        return prevExtract;
+      }, null);
+  }
+
+  componentDidMount() {
+    this.callForceUpdate = this.callForceUpdate.bind(this);
+    document.addEventListener('rowHeightRecomputed', this.callForceUpdate);
+  }
+
+  componentDidUpdate() {
+    const { fullLevel, rowIndex, postId } = this.props;
+    const extracts = [].slice.call(document.getElementsByClassName('extracts'));
+    const prevExtract = Nuggets.previousExtract(Nuggets.completeLevel(rowIndex, fullLevel), extracts);
+    if (prevExtract !== null) {
+      const newTop = getDomElementOffset(prevExtract).top + prevExtract.getBoundingClientRect().height + Nuggets.SPACER_SIZE;
+      const postTop = getDomElementOffset(document.getElementById(postId)).top;
+      setDomElementOffset(this.node, { top: Math.max(postTop, newTop) });
+    }
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('rowHeightRecomputed', this.callForceUpdate);
+  }
+
+  callForceUpdate() {
+    // This is to prevent the event from passing arguments to forceUpdate, raising an error.
+    // I did not use an arrow function because we need to pass the function to removeEventListener
+    this.forceUpdate();
```

Comment:
Another approach... Instead of the `forceUpdate` you can do that:

```
componentDidMount() {
    this.setState({rowHeight: height});
    window.addEventListener('resize', rowHeightRecomputed);
}
componentWillUnmount() {
    window.removeEventListener('resize', rowHeightRecomputed);
  }
```

Reply:
The row height is not used, the nuggets component needs to rerender whenever a post is loaded which is what the rowHeightRecomputed event signals (there is no rowHeightRecomputed function)

Reply:
I found a way to not use forceUpdate